From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Fri, 18 Jul 2026 01:25:00 +0000
Subject: arm64: dts: allwinner: t527: avaota-a1: enable WiFi and Bluetooth

The Avaota A1 board has an AIC8800 WiFi/BT combo chip connected via
SDIO (SDC1 on PG0-PG5) and UART1 (PG6-PG9). Enable the MMC1 controller
with a simple power sequence for the WL-REG-ON pin (PG13), using DCDC4
(3.3V, VCC-3V3) as vmmc (VBAT) and BLDO1 (1.8V, VCCIO-WIFI / VCC-PG)
as vqmmc (I/O signaling). Enable UART1 with RTS/CTS for the Bluetooth
side, with shutdown (PG12), host-wakeup (PG10) and device-wakeup (PG11)
GPIOs. Mark reg_bldo1 as always-on since it powers VCC-PG (PG port for
SDIO, UART1, control GPIOs) and VCCIO-WIFI (AIC8800 VDDIO and BT
pull-ups).

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
Assisted-by: opencode:big-pickle
---
diff --git a/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts b/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts
index 994f1e4..5645a35 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts
@@ -6,6 +6,7 @@
 #include "sun55i-a523.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/leds/common.h>
 
 / {
@@ -16,6 +17,7 @@
 		ethernet0 = &gmac0;
 		ethernet1 = &gmac1;
 		serial0 = &uart0;
+		serial1 = &uart1;
 	};
 
 	chosen {
@@ -63,6 +65,12 @@
 		gpio = <&pio 8 12 GPIO_ACTIVE_HIGH>;	/* PI12 */
 		enable-active-high;
 	};
+
+	wifi_pwrseq: wifi-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&pio 6 13 GPIO_ACTIVE_LOW>;	/* PG13 WL-REG-ON */
+		post-power-on-delay-ms = <200>;
+	};
 };
 
 &ehci0 {
@@ -173,6 +181,25 @@
 	status = "okay";
 };
 
+&mmc1 {
+	bus-width = <4>;
+	max-frequency = <40000000>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	non-removable;
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc1_pins>;
+	vmmc-supply = <&reg_dcdc4>;
+	vqmmc-supply = <&reg_bldo1>;
+	status = "okay";
+
+	wifi@1 {
+		reg = <1>;
+		interrupt-parent = <&pio>;
+		interrupts = <6 14 IRQ_TYPE_LEVEL_LOW>;	/* PG14 WL-WAKE-AP */
+		interrupt-names = "host-wake";
+	};
+};
+
 &mmc2 {
 	bus-width = <8>;
 	cap-mmc-hw-reset;
@@ -289,6 +316,8 @@
 			};
 
 			reg_bldo1: bldo1 {
+				/* VCC-PG (PG port) + VCCIO-WIFI (AIC8800 VDDIO + BT pull-ups) */
+				regulator-always-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-name = "vcc-pg-wifi-lvds";
@@ -430,6 +459,20 @@
 	status = "okay";
 };
 
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1_pins>, <&uart1_rts_cts_pins>;
+	uart-has-rtscts;
+	status = "okay";
+
+	bluetooth {
+		/* aic8800-bt */
+		shutdown-gpios = <&pio 6 12 GPIO_ACTIVE_LOW>;	/* PG12 BT-RESETN */
+		host-wakeup-gpios = <&pio 6 10 GPIO_ACTIVE_HIGH>;	/* PG10 BT-HOST-WAKE */
+		device-wakeup-gpios = <&pio 6 11 GPIO_ACTIVE_HIGH>;	/* PG11 AP-WAKE-BT */
+	};
+};
+
 &usb_otg {
 	 /*
 	  * The CC pins of the USB-C port have two pull-down resistors
